### PR TITLE
Lsig Web Support Tutorial

### PR DESCRIPTION
**Pull request by Megha-Dev-19**

### CODE REVIEW

File: docs/tutorials/t-10.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+The purpose of this tutorial is to demonstrate the usage of Algo Builder Web Package to sign and send a Logic Signature Transaction using major Algorand wallets (Algo Signer, Wallet Connect, MyAlgoWallet).
+
+### Requirements
+
+- [Algo Builder](https://algobuilder.dev/) and it's [dependencies](https://algobuilder.dev/guide/README#requirements)
+- Setup the accounts on [`Algosigner`](https://github.com/PureStake/algosigner), [`Wallet Connect`](https://walletconnect.com/) wallet and [`MyAlgo Wallet`](https://wallet.myalgo.com/). It could be MainNet, TestNet or private net.
+
+## Steps
+
+In this tutorial we will use Algo Builder Wallet Widget to connect to various web wallets. Then we will show how to instrument the wallet to sign a multisignature transaction. See [Wallet Widget tutorial](https://developer.algorand.org/solutions/vuejs-multi-wallet-widget/) for more details about the widget.
```
Comment (robert-zaremba) [inline docs/tutorials/t-10.md:12]: wrong copy paste. Please update this section accordingly.

File: docs/tutorials/t-10.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	closeToRemaninder,
+	note,
+	params
+);
+```
+
+After creating the transaction, you can sign it using different wallets.
+
+1. AlgoSigner<br/>
+   We use `connectAlgoSigner()` method to connect the AlgoSigner wallet. Next, we create an instance of the `WebMode` class by passing `AlgoSigner` and the Chain Name i.e the network name. The `web` instance will allow us to sign and send transactions to the blockchain.
```
Comment (robert-zaremba) [inline docs/tutorials/t-10.md:39]:
```suggestion
#### AlgoSigner

Use `connectAlgoSigner()` method to connect the AlgoSigner wallet. Next, we create an instance of the `WebMode` class by passing `AlgoSigner` and the Chain Name i.e the network name. The `web` instance will allow us to sign and send transactions to the blockchain.
```

File: docs/tutorials/t-10.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+```js
+const web = new WebMode(AlgoSigner, CHAIN_NAME);
+```
+
+Webmode method `signLogicSignatureTxn` which is used to sign the txn, expects transaction and LogicSigAccount as parameters.
+To sign the lsig txn:
```
Comment (robert-zaremba) [inline docs/tutorials/t-10.md:46]:
```suggestion
Use `web.signLogicSignatureTxn` method to sign the txn. It expects transaction and LogicSigAccount as arguments:
```

File: docs/tutorials/t-10.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+```js
+const signedTxnResponse = await web.signLogicSignatureTxn(lsigTxn, lsig);
+```
+
+After signing, the returned response is the object containing blob key, which we can further use to send the signed transaction to the network.
```
Comment (robert-zaremba) [inline docs/tutorials/t-10.md:52]:
```suggestion
After signing, the returned object contains signed transaction data in a `blob` field, which we can use to submit to the network.
```

File: docs/tutorials/t-10.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+To send the above txn we need to first convert it to base64:
+
+```js
+const base64Txn = Buffer.from(signedTxnResponse.blob).toString("base64");
+const sendTxnResponse = await webmode.sendAndWait(base64Txn);
+```
+
+The sent txn response includes the confirmed transaction response along with txID.
```
Comment (robert-zaremba) [inline docs/tutorials/t-10.md:60]:
```suggestion
The `sendTxnResponse` includes the confirmed transaction response along with txID.
```

File: docs/tutorials/t-10.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+2. MyAlgo Wallet <br/>
+   We use `connectMyAlgoWallet()` method to connect to My Algo wallet.
```
Comment (robert-zaremba) [inline docs/tutorials/t-10.md:63]:
```suggestion
#### MyAlgo Wallet

Let's create `connectMyAlgoWallet()` function to connect to My Algo wallet:
```

File: docs/tutorials/t-10.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+        }
+        let myAlgoMode = new MyAlgoWalletSession(walletURL);
+        await myAlgoMode.connectToMyAlgo();
+        if (myAlgoMode.accounts.length) {
+          this.walletAddress = myAlgoMode.accounts[0].address;
+        }
+      } catch (e) {
+        console.error(e);
+      }
+   }
```
Comment (robert-zaremba) [inline docs/tutorials/t-10.md:81]: why we don't have this function in the `WebMode`?

File: docs/tutorials/t-10.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+```
+
+MyAlgo Wallet method `signLogicSignatureTxn` also works similar to Algosigner webmode.
+To send the signed txn returned from MyAlgo Wallet:
+
+```js
+const sendTxnResponse = await algoWallet.sendAndWait(signedTxnResponse.blob);
+```
+
+Here also the sent txn response includes the confirmed transaction response along with txID.
```
Comment (robert-zaremba) [inline docs/tutorials/t-10.md:91]:
```suggestion
Here also the `sendTxnResponse` includes the confirmed transaction response along with txID.
```

File: docs/tutorials/t-10.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+3. Wallet Connect<br/>
+   We use `connectWalletConnect()` method is to connect to Wallet Connect wallet.
```
Comment (robert-zaremba) [inline docs/tutorials/t-10.md:94]:
```suggestion
#### Wallet Connect

Let's create  a `connectWalletConnect()` method is to connect to a Wallet Connect session.
```

File: docs/tutorials/t-10.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+        await walletConnector.create(true);
+        walletConnector.onConnect((error, response) => {
+          if (response.accounts.length) {
+            this.walletAddress = response.accounts[0];
+          }
+        });
+      } catch (e) {
+        console.error(e);
+      }
+   }
```
Comment (robert-zaremba) [inline docs/tutorials/t-10.md:114]: again, why we don't have it in the `WebMode`?

File: docs/tutorials/t-10.md (hunk 7 above)
Comment (Megha-Dev-19) [reply docs/tutorials/t-10.md:82]: Do you mean catching the error and throwing it?

File: docs/tutorials/t-10.md (hunk 1 above)
Comment (Megha-Dev-19) [reply docs/tutorials/t-10.md:12]: oh yes, will update

File: docs/tutorials/t-10.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
+# Algo Builder Web Support for Logic Signature Transaction
+
+The purpose of this tutorial is to demonstrate the usage of Algo Builder Web Package to sign and send a Logic Signature Transaction using major Algorand wallets (Algo Signer, Wallet Connect, MyAlgoWallet).
```
Comment (robert-zaremba) [inline docs/tutorials/t-10.md:3]:
```suggestion
# Algo Builder Web Support for Smart Signature Transaction

The purpose of this tutorial is to demonstrate the usage of Algo Builder Web Package to sign and send a Smart Signature Transaction (lsig) using major Algorand wallets (Algo Signer, Wallet Connect, MyAlgoWallet).
```

File: docs/tutorials/t-10.md [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+### Requirements
+
+- [Algo Builder](https://algobuilder.dev/) and it's [dependencies](https://algobuilder.dev/guide/README#requirements)
+- Setup the accounts on [`Algosigner`](https://github.com/PureStake/algosigner), [`Wallet Connect`](https://walletconnect.com/) wallet and [`MyAlgo Wallet`](https://wallet.myalgo.com/). It could be MainNet, TestNet or private net.
+
+## Steps
+
+In this tutorial we will use Algo Builder Wallet Widget to connect to various web wallets. Then we will show how to instrument the wallet to sign a Logic Signature transaction. See [Wallet Widget tutorial](https://developer.algorand.org/solutions/vuejs-multi-wallet-widget/) for more details about the widget.
+
+### LogicSignature Transaction
```
Comment (robert-zaremba) [inline docs/tutorials/t-10.md:14]:
```suggestion
### Smart Signature Transaction
```

File: docs/tutorials/t-10.md (hunk 7 above)
Comment (robert-zaremba) [reply docs/tutorials/t-10.md:82]: nevermind, probably it's no worth

**Review by robert-zaremba — COMMENTED** [review]
Let's update this tutorial after finalizing https://github.com/scale-it/algo-builder/pull/846#pullrequestreview-1199005392

**Comment by Megha-Dev-19** [comment]
> Let's update this tutorial after finalizing [#846 (review)](https://github.com/scale-it/algo-builder/pull/846#pullrequestreview-1199005392)

Yeah I have updated the method names.

**Review by robert-zaremba — APPROVED** [review]

**Pull request merged**